ui/components/app/confirm/info/row/alert-row.tsx

```diff
@@ -47,6 +48,7 @@ export const AlertRow = ({
   tooltip,
   variant = ConfirmInfoRowVariant.Default,
 }: AlertRowProps) => {
+  const processAlertAction = useConfirmationAlertActions();
```

Comment:
Unfortunately we can't couple the confirmation actions to this alert row since it's not confirmations specific hence the `ownerId` property.

We could provide an `onActionClick` property here but then we'd have to pass it to every row which would be awkward and create lots of duplication.

Perhaps a good compromise is to create a new React context such as `AlertActionHandler` that provides a `processAction` function.

We could then use that directly within the `AlertModal` so we don't need to pass around handlers, then we can just add a single `AlertActionHandlerProvider` higher up the stack where we're already coupled to confirmations, which can use our `useConfirmationAlertActions` hook.

Reply:
Perfect, that is a great suggestion. I created the new react context and started using it on the `MultipleAlertModal`  providing the `processAction` and accessing it in the `AlertModal`.